### DISCUSSION TITLE
[@types/pkijs]  Property 'importKey' in type 'CryptoEngine' is not assignable to the same property in base type 'SubtleCrypto'.

### DISCUSSION THREAD

**Discussion by dninet** [DefinitelyTyped/DefinitelyTyped#62420] [Issues with a @types package]
This is an issue with the `@types/pkijs` package although this also happens when I have the `pkijs` package installed.
```bash
node_modules/@types/pkijs/index.d.ts(636,9): error TS2416: Property 'importKey' in type 'CryptoEngine' is not 
assignable to the same property in base type 'SubtleCrypto'.
[build:dev]   Type '{ (format: "jwk", keyData: JsonWebKey, algorithm: string | HmacImportParams | 
RsaHashedImportParams | EcKeyImportParams, extractable: boolean, keyUsages: readonly string[]): 
Promise<...>; (format: "spki" | ... 1 more ... | "raw", keyData: BufferSource, algorithm: string | ... 2 more ... | 
EcKeyImportParams, extracta...' is not assignable to type '{ (format: "jwk", keyData: JsonWebKey, algorithm: 
AlgorithmIdentifier | HmacImportParams | RsaHashedImportParams | EcKeyImportParams | AesKeyAlgorithm, 
extractable: boolean, keyUsages: readonly KeyUsage[]): Promise<...>; (format: "spki" | ... 1 more ... | "raw", 
keyData: BufferSource, algorithm: AlgorithmIdentifier ...'.
[build:dev]     Types of parameters 'format' and 'format' are incompatible.
[build:dev]       Type '"spki" | "pkcs8" | "raw"' is not assignable to type '"jwk"'.
[build:dev]         Type '"spki"' is not assignable to type '"jwk"'.
```

I get this error (and 11 other similar errors) when I am transpiling my TS project to JS. I have not found a similar issue with the exception of [this page](https://github.com/PeculiarVentures/webcrypto-liner/issues/60)

[a comment by an automated account is not shown]

**Comment by dninet** [comment]
Well, I tried out a few things and found out that if I don't import it using conventional `import pkijs from "@types/pkijs"` but instead use the old `const pkijs = require("pkijs")`, it works fine. This is weird.

**Comment by Semigradsky** [comment]
@AlbertDavid94 As I see, pkijs has its own typings. So just try to remove `@types/pkijs`, looks like this package is no needed.

**Reply by dninet** [reply]
I've tried this as well, it didn't work.

**Reply by Semigradsky** [reply]
What error do you have after removing `@types/pkijs` and reinstalling dependencies?

**Reply by dninet** [reply]
When I remove `@types/pkijs` from the `packages.json` and `npm install pkijs`, I get the same 11 errors.

**Reply by Semigradsky** [reply]
Remove `node_modules` before installing dependencies.
